pysat/tests/classes/cls_instrument_iteration.py

```diff
@@ -0,0 +1,789 @@
+                                                by_date):
+        """Test iteration via date season step/width>1, exclude stop date."""
+
+        out = self.support_iter_evaluations(values, reverse=reverse,
+                                            by_date=by_date)
+        self.verify_iteration(out, reverse=reverse, inclusive=False)
+
+        return
+
+    @pytest.mark.parametrize("new_bounds,errmsg",
+                             [([dt.datetime(2009, 1, 1)],
+                               "Must supply both a start and stop date/file"),
+                              ([dt.datetime(2009, 1, 1), '2009-01-01.nofile'],
+                               "must all be of the same type"),
+                              ([dt.datetime(2009, 1, 1), 1],
+                               "must all be of the same type"),
+                              ([[dt.datetime(2009, 1, 1)] * 2,
+                                '2009-01-01.nofile'],
+                               "must have the same number of elements"),
+                              ([[dt.datetime(2009, 1, 1)] * 2,
+                               [dt.datetime(2009, 1, 1), '2009-01-01.nofile']],
+                               "must all be of the same type"),
+                              ([dt.datetime(2009, 1, 1),
+                                dt.datetime(2009, 1, 1), '1D',
+                                dt.timedelta(days=1), False],
+                               'Too many input arguments.')])
+    def test_set_bounds_error_message(self, new_bounds, errmsg):
+        """Test ValueError when setting bounds with wrong inputs."""
+
+        with pytest.raises(ValueError) as verr:
+            self.testInst.bounds = new_bounds
+        assert str(verr).find(errmsg) >= 0
+        return
+
+    def test_set_bounds_string_default_start(self):
+        """Test set bounds with default start."""
+
+        self.testInst.bounds = [None, '2009-01-01.nofile']
+        assert self.testInst.bounds[0][0] == self.testInst.files[0]
+        return
+
+    def test_set_bounds_string_default_stop(self):
+        """Test set bounds with default stop."""
+
+        self.testInst.bounds = ['2009-01-01.nofile', None]
+        assert self.testInst.bounds[1][0] == self.testInst.files[-1]
+        return
+
+    def test_set_bounds_by_default_dates(self):
+        """Verify bounds behavior with default date related inputs."""
+
+        start = self.testInst.files.start_date
+        stop = self.testInst.files.stop_date
+        full_list = pds.date_range(start, stop).tolist()
+        self.testInst.bounds = (None, None)
+        assert np.all(self.testInst._iter_list == full_list)
+        self.testInst.bounds = None
+        assert np.all(self.testInst._iter_list == full_list)
+        self.testInst.bounds = (start, None)
+        assert np.all(self.testInst._iter_list == full_list)
+        self.testInst.bounds = (None, stop)
+        assert np.all(self.testInst._iter_list == full_list)
```

Comment:
Add comments explaining what each assert statement is doing.

Reply:
Switching to use the standard evaluation method.